opendbc/car/hyundai/carstate.py

```diff
@@ -168,11 +169,15 @@ def update(self, cp, cp_cam, *_) -> structs.CarState:
     self.clu11 = copy.copy(cp.vl["CLU11"])
     self.steer_state = cp.vl["MDPS12"]["CF_Mdps_ToiActive"]  # 0 NOT ACTIVE, 1 ACTIVE
     prev_cruise_buttons = self.cruise_buttons[-1]
+    prev_lfa_button = self.lfa_button
     self.cruise_buttons.extend(cp.vl_all["CLU11"]["CF_Clu_CruiseSwState"])
     self.main_buttons.extend(cp.vl_all["CLU11"]["CF_Clu_CruiseSwMain"])
+    self.lfa_button = cp.vl["BCM_PO_11"]["LFA_Pressed"] if self.CP.flags & HyundaiFlags.HAS_LFA_BUTTON else 0
 
-    if self.CP.openpilotLongitudinalControl:
-      ret.buttonEvents = create_button_events(self.cruise_buttons[-1], prev_cruise_buttons, BUTTONS_DICT)
+    ret.buttonEvents = [
+      *(create_button_events(self.cruise_buttons[-1], prev_cruise_buttons, BUTTONS_DICT) if self.CP.openpilotLongitudinalControl else []),
+      *create_button_events(self.lfa_button, prev_lfa_button, {1: ButtonType.altButton1})
+    ]
```

Comment:
This is less lines than an if statement, but I'd argue harder to quickly read

Reply:
How about something like this?
```
if self.CP.openpilotLongitudinalControl:
  ret.buttonEvents = create_button_events(self.cruise_buttons[-1], prev_cruise_buttons, BUTTONS_DICT)

ret.buttonEvents = [
  *ret.buttonEvents,
  *create_button_events(self.lfa_button, prev_lfa_button, {1: ButtonType.altButton1})
]